src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,98 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: {
+    todos: Todo[];
+    originalTodos: Todo[];
+    searchQueryTodo: Todo[];
+  },
+  setTodos: React.Dispatch<React.SetStateAction<{
+    todos: Todo[];
+    originalTodos: Todo[];
+    searchQueryTodo: Todo[];
+  }>>,
+};
+
+export const TodoFilter: React.FC<Props> = ({ todos, setTodos }) => {
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(e.target.value);
+    const newQuery = todos.searchQueryTodo.filter(todo => todo.title
+      .toLowerCase().includes(e.target.value.trim().toLowerCase()));
+
+    setTodos(prev => ({
+      ...prev,
+      todos: newQuery,
+    }));
+  };
+
+  const handleClearSearch = () => {
+    setSearchQuery('');
+  };
+
+  const handleFilterTodos = (filterType: string) => {
+    let filteredTodos = todos.originalTodos;
+
+    switch (filterType) {
+      case 'active':
+        filteredTodos = todos.originalTodos.filter(todo => !todo.completed);
+        break;
+      case 'completed':
+        filteredTodos = todos.originalTodos.filter(todo => todo.completed);
+        break;
+      default:
+    }
+
+    setTodos(prev => ({
+      ...prev,
+      todos: filteredTodos,
+      searchQueryTodo: filteredTodos,
+    }));
+  };
```

Comment:
move this code to App.tsx. in this component would be better just to show and change filters